plugin/helpers_bots.go

```diff
@@ -72,3 +117,49 @@ func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error)
 
 	return createdBot.UserId, nil
 }
+
+func (p *HelpersImpl) ShouldProcessMessage(post *model.Post, options ...ShouldProcessMessageOption) (bool, error) {
+	messageProcessOptions := &shouldProcessMessageOptions{}
+	for _, option := range options {
+		option(messageProcessOptions)
+	}
+
+	if post.IsSystemMessage() && !messageProcessOptions.AllowSystemMessages {
+		return false, nil
+	}
+
+	if !messageProcessOptions.AllowBots {
+		user, appErr := p.API.GetUser(post.UserId)
+		if appErr != nil {
+			return false, errors.Wrap(appErr, "unable to get user")
+		}
+
+		if user.IsBot {
+			return false, nil
+		}
+	}
```

Comment:
I'd suggest that even if we `AllowBots`, we should ignore messages from our own bot, since that's very likely to cause an infinite loop. So basically move up the `BOT_USER_KEY` checks into an `else` condition here. 

I don't think this needs to be configurable. It might be interesting to have an option that is, `OnlyBotDMs` that effectively does what you do below, but in general we want to be able to process messages from other contexts too.